workers/unity/Assets/Fps/Scripts/SetupLogic/SimulatedPlayerWorkerConnector.cs

```diff
@@ -1,18 +1,86 @@
 {
     private const string AuthPlayer = "Prefabs/SimulatedPlayer/SimulatedPlayer";
     private const string NonAuthPlayer = "Prefabs/SimulatedPlayer/SimulatedPlayerProxy";
 
-    private async void Start()
+    private ILogDispatcher simulatedCoordinatorLogDispatcher;
+    private bool connectToRemoteDeployment;
+    private string loginToken;
+    private string playerIdentityToken;
+
+    public async Task ConnectSimulatedPlayer(ILogDispatcher logDispatcher, string SimulatedPlayerDevAuthTokenId, string SimulatedPlayerTargetDeployment)
     {
-        await Connect(WorkerUtils.SimulatedPlayer, new ForwardingDispatcher());
+        simulatedCoordinatorLogDispatcher = logDispatcher;
+
+        // If we're connecting simulated players to another deployment, we need to fetch locator tokens.
+        if (!string.IsNullOrEmpty(SimulatedPlayerTargetDeployment))
+        {
+            if (string.IsNullOrEmpty(SimulatedPlayerDevAuthTokenId))
+            {
+                logDispatcher.HandleLog(LogType.Error, new LogEvent("Failed to launch simulated player. No development authentication token specified."));
+                return;
+            }
+
+            // Obtain player identity token for simulated player.
+            var playerIdentityTokenResponse = DevelopmentAuthentication.CreateDevelopmentPlayerIdentityTokenAsync("locator.improbable.io", 444,
+                new PlayerIdentityTokenRequest
+                {
+                    DevelopmentAuthenticationTokenId = SimulatedPlayerDevAuthTokenId,
+                    DisplayName = "",
+                    PlayerId = "SimulatedPlayer"
+                }).Get();
+            if (playerIdentityTokenResponse.Value.Status != ConnectionStatusCode.Success)
+            {
+                logDispatcher.HandleLog(LogType.Error, new LogEvent("Failed to launch simulated player. Failed to obtain player identity token. Reason: " + playerIdentityTokenResponse?.Error));
+                return;
+            }
+            playerIdentityToken = playerIdentityTokenResponse?.PlayerIdentityToken;
+
+            // Obtain login token for simulated player.
+            var loginTokenResponse = DevelopmentAuthentication.CreateDevelopmentLoginTokensAsync("locator.improbable.io", 444,
+                new LoginTokensRequest
+                {
+                    PlayerIdentityToken = playerIdentityToken,
+                    WorkerType = WorkerUtils.UnityClient,
+                    DurationSeconds = 600
+                }).Get();
+            if (loginTokenResponse.Value.Status != ConnectionStatusCode.Success)
+            {
+                logDispatcher.HandleLog(LogType.Error, new LogEvent("Failed to launch simulated player. Failed to obtain login tokens. Reason: " + loginTokenResponse?.Error));
+                return;
+            }
+            if (loginTokenResponse?.LoginTokens != null)
+            {
+                foreach (var loginTokenDetails in loginTokenResponse.Value.LoginTokens)
```

Comment:
This block can be made simpler with some LINQ

Reply:
Done 👍